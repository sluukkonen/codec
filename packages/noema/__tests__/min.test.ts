import { min, number } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

describe("min", () => {
  test("should parse if value is greater than or equal to the minimum", () => {
    const nonNegativeNumber = min(number, 0)
    expectParseSuccess(nonNegativeNumber, 0)
    expectParseSuccess(nonNegativeNumber, 1)
  })

  test("should fail to parse if value is less than the minimum", () => {
    const nonNegativeNumber = min(number, 0)
    expectParseFailure(nonNegativeNumber, -1, [
      { code: "too_small", path: [], min: 0 },
    ])
  })

  test("should fail to parse if value isn't parsed by the underlying codec", () => {
    const nonNegativeNumber = min(number, 0)
    expectParseFailure(nonNegativeNumber, "", [
      { code: "invalid_number", path: [] },
    ])
  })
})
