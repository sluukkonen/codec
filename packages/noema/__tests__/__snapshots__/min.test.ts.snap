// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`min should fail to parse if value is less than the minimum 1`] = `
[
  {
    "code": "too_small",
    "min": 0,
    "path": [],
  },
]
`;

exports[`min should fail to parse if value isn't parsed by the underlying codec 1`] = `
[
  {
    "code": "invalid_number",
    "path": [],
  },
]
`;
